### ISSUE TITLE
Examples: Inaccessible ad content

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31182]
## WCAG Level
Level A, AA

## Priority
Medium

## Type
Various

## Pages/screens/components affected
* [amp-sticky-ad](https://amp.dev/documentation/components/amp-sticky-ad/)
* [amp-auto-ads](https://amp.dev/documentation/components/amp-auto-ads/)

## Description
While not under the control of the AMP project itself, nor of authors using AMP, we noted during the assessment that many of the third-party adverts that were embedded into the sample pages for the `amp-sticky-ad` and `amp-auto-ads` components exhibit various accessibility issues. These ranged from images that lacked alternative text, to low contrast for text and non-text content, links lacking any kind of link text and accessible name.

Example of an advert with no text alternative, resulting in a lack of link text and a lack of accessible name for the control itself:

!['Google for Work | Chrome' sticky ad in example page](https://user-images.githubusercontent.com/54933393/99182205-754d2880-272b-11eb-8543-300247f9a2d3.png)

```
<a id="aw0" target="_blank" href="https://googleads.g.doubleclick.net ..." onfocus="..." onmousedown="..." onmouseover="..." onclick="...">
  <img src="..." border="0" width="320" height="50" alt="" class="img_ad">
</a>
```

Example of controls that lack an accessible name and present insufficient non-text color contrast - using a teal `#00AECD` foreground color on a gray `#CDCCCC` background, with a resulting color contrast ratio of 1.7:1, below the minimum 3:1:

![AdChoices buttons](https://user-images.githubusercontent.com/54933393/99182208-7d0ccd00-272b-11eb-93aa-1089c29755d4.png)

## User impact
Various user groups will be impacted by inaccessible adverts, depending on the specific problems that individual ads present.

When images do not provide a text alternative, users who cannot perceive images (including users of assistive technologies) will not be able to understand the meaning/content of the adverts.

Links and controls that lack an appropriate accessible name will be announced by assistive technologies simply as "link" or "button", conveying no context or purpose to users.

Text and graphical content with insufficient color contrast will make it difficult or impossible for users with impaired color perception to correctly read content or operate controls.

## Required solution
As the adverts themselves, as well as any interface they may be present, are outside of the control of the AMP project, the only solution would be to flag the issues with the ad content providers (such as AdChoices), to ensure that the markup and content that is injected into pages as a result of an ad (the ad itself, as well as any controls that are generated) is accessible.

## Related standards
* [WCAG 2.1 Success Criterion 1.1.1 Non-text Content (Level A)](https://www.w3.org/TR/WCAG21/#non-text-content)
* [WCAG 2.1 Success Criterion 2.4.4 Link Purpose (In Context) (Level A)](https://www.w3.org/TR/WCAG21/#link-purpose-in-context)
* [WCAG 2.1 Success Criterion 2.5.3 Label in Name (Level A)](https://www.w3.org/TR/WCAG21/#label-in-name)
* [WCAG 2.1 Success Criterion 4.1.2 Name, Role, Value (Level A)](https://www.w3.org/TR/WCAG21/#name-value)

## Test data
Test date: November 2020
